docs/text-tile-plugins.md

```diff
@@ -11,3 +11,70 @@ A text tile plugin has to provide the following info:
+
+A better design would be to allow a single plugin registration to include multiple tools.
+And also the name `modalHook` is too specific a plugin tool might just be changing the state of the slate without bringing up a modal like a bold button.
+
+Each of these tools needs:
+- Icon
+- toolName
+- toolTip
+- command to run when clicked
+- TODO: someway of handling disabling the button depending on where the cursor is
+- TODO: someway of highlighting the button depending on where the cursor is like what bold and italics buttons do
+
+If we rename iconName to toolName then we might need a top level name to identify the plugin, but I haven't found a reason yet.
+
+What is needed for the selected buttons support is a map from slate type to the button id.
+This could be added to each of the tool definitions from the plugin.
+
+These slate types are added here:
+...getTextPluginIds().reduce((idMap, id) => ({...idMap, [id]: id}), {})
+
+This is in the text-tile.tsx `getSelectedIcons` that slateToButtonType is used to compute the selected toolbar buttons. They are updated each time the content changes. And they are then passed into the TextToolbar.
+
+What is weird about that is that the TextToolbar does its own isElementActive check for the variables. The getSelectedIcons uses isMarkActive and isBlockActive. Since different buttons seem to use different checks. It might work for each button to just provide its own check. The check could return if the button is selected or enabled. This way the button doesn't have to provide any slate types since that would all be internal to the button.
+
+For the built in buttons they use EFormat.
+
+This is currently handled by a buttonEnabled function on text plugin infos which gets a variable (which it shouldn't).
+It isn't clear if this should be done just by including a slateType on each button definition, or if we need more.
+The trick is that for things like Bold it doesn't become disabled it is just marked as selected. So these
+variable dialog buttons are different in that they become disabled (like a link edit button would be) if the thing selected
+is not the right thing. For a link you'd want it to become enabled when any text is selected not just the link type.
+
+So to support more kinds of tools it can't just be based on the slate type. If we could leave it up to the tool to decide when it should be selected and when it should be enabled based on the current selection and mark that would be best...
+
+It turns out that editor.isElementActive(type) is the same as isBlockActive(editor, type)
+See create-editor.ts in slate-editor repo. This editor.isElementActive is our function from slate-editor it finds all of the nodes associated with the current selection and by defaults checks if any node.type == type.
+
+We can't use the Cypress tests easily because of some issue with CLUE, Slate, and Cypress
+
+TODO:
+I'd note that the slate-editor defines a EFormat.variable which it shouldn't be doing.
```

Comment:
It probably isn't worth reading this. It is just a running list of notes not actually documentation. Some of the notes are wrong now. I'll clean this up before requesting a review.